pycloudlib/openstack/instance.py

```diff
@@ -156,3 +164,40 @@ def wait_for_delete(self):
     def wait_for_stop(self):
         """Wait for instance stop."""
         self.conn.compute.wait_for_server(self.server, status='SHUTOFF')
+
+    def add_network_interface(self) -> str:
+        """Add nic to running instance.
+
+        Returns IP address in string form
+        """
+        port = self.conn.network.create_port(
+            network_id=self.network_id,
+            node_id=self.server.id
+        )
+        self.added_local_ports.append(port.id)
+        interface = self.conn.compute.create_server_interface(
+            server=self.server.id,
+            port_id=port.id
+        )
+        return interface['fixed_ips'][0]['ip_address']
```

Comment:
Same thing here regarding logging a better error message, if the error can happen of course

Reply:
If the line above fails in some way, it should raise it's own error. If it succeeds, then these fields should all be present. If they're not present for some reason, then we'll need to dive in and debug why not, and I don't think any additional exception wrapping would be advantageous in that case.